src/Offer/CommandHandlers/ImportLabelsHandler.php

```diff
@@ -37,20 +49,88 @@ public function handle($command): void
             return;
         }
 
-        /** @var Udb3ModelLabel $importLabel */
-        foreach ($command->getLabelsToImport() as $importLabel) {
+        $offer = $this->offerRepository->load($command->getItemId());
+
+        $labelsToImport = $command->getLabels();
+        $labelsToImport = $this->fixVisibility($labelsToImport);
+
+        $labelsOnOffer = new Labels();
+        foreach ($offer->getLabels()->asArray() as $labelOnOffer) {
```

Comment:
Since `Offer::getLabels()` is new since this PR, maybe it would be best to do the conversion of a `LabelCollection` to `Labels` in there? That way you don't need to refactor how `$labels` is stored in `Offer` right now but the `getLabels()` method can already consistently return the new `Labels` VO. And it would make this relatively big `handle()` method a little bit smaller and easier to understand.

Reply:
Will be picked up in https://jira.uitdatabank.be/browse/III-4620

Reply:
The method `Offer::getLabels` now returns a VO `Labels`